tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -318,6 +318,19 @@ def test_band_data(self):
         self.assertEqual(self.band.datatype(), 1)
         self.assertEqual(self.band.datatype(as_string=True), 'GDT_Byte')
         self.assertEqual(self.band.nodata_value, 15)
+        if numpy:
+            data = self.band.data()
+            # Check that the 42 first values of the column 42 match what's
+            # inside the file raster.tif
+            correct_list = [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+                            4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 3, 3, 9, 9,
+                            9, 4, 4, 4, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4]
+            for x in range(0, 42):
+                self.assertEqual(data[x, 42], correct_list[x])
+            self.assertEqual(len(data), self.band.height)
+            for row in data:
```

Comment:
Similarly here, you could use the `data.shape` attribute to compare directly `assertEqual(data.shape, (self.band.height, self.band.width)`.